src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -77,74 +77,89 @@ public async Task<JObject> GetValueFromObject(JToken objRet, CancellationToken t
-                    string.IsNullOrEmpty(methodInfo?.TypeInfo?.Namespace) ?
-                    classNameToFind :
-                    methodInfo.TypeInfo.Namespace + "." + classNameToFind;
-
-                var searchResult = await TryFindNameInAssembly(store.assemblies, classNameToFindWithNamespace);
-                if (searchResult == null)
-                    searchResult = await TryFindNameInAssembly(store.assemblies, classNameToFind);
-                if (searchResult != null)
-                    typeId = (int)searchResult;
-
-                async Task<int?> TryGetTypeIdFromName(string typeName, AssemblyInfo assembly)
+
+                if (!string.IsNullOrEmpty(methodInfo.TypeInfo.Namespace))
                 {
-                    var type = assembly.GetTypeByName(typeName);
-                    if (type == null)
-                        return null;
-                    return await context.SdbAgent.GetTypeIdFromToken(assembly.DebugId, type.Token, token);
+                    typeId = await FindStaticTypeId(methodInfo.TypeInfo.Namespace + "." + classNameToFind);
+                    if (typeId != -1)
+                        continue;
                 }
+                typeId = await FindStaticTypeId(classNameToFind);
+            }
+
+            return (null, null);
 
-                async Task<int?> TryFindNameInAssembly(List<AssemblyInfo> assemblies, string name)
+            async Task<JObject> FindStaticMemberInType(string name, int typeId)
+            {
+                var fields = await context.SdbAgent.GetTypeFields(typeId, token);
+                foreach (var field in fields)
                 {
-                    foreach (var asm in assemblies)
+                    if (field.Name != name)
+                        continue;
+
+                    var isInitialized = await context.SdbAgent.TypeIsInitialized(typeId, token);
+                    if (isInitialized == 0)
                     {
-                        var typeId = await TryGetTypeIdFromName(name, asm);
-                        if (typeId != null)
-                            return typeId;
+                        isInitialized = await context.SdbAgent.TypeInitialize(typeId, token);
                     }
-                    return null;
+                    var valueRet = await context.SdbAgent.GetFieldValue(typeId, field.Id, token);
+
+                    return await GetValueFromObject(valueRet, token);
+                }
+
+                var methodId = await context.SdbAgent.GetPropertyMethodIdByName(typeId, name, token);
+                if (methodId != -1)
+                {
+                    var commandParamsObj = new MemoryStream();
+                    var commandParamsObjWriter = new MonoBinaryWriter();
+                    commandParamsObjWriter.Write(0); //param count
+                    var retMethod = await context.SdbAgent.InvokeMethod(commandParamsObj.ToArray(), methodId, "methodRet", token);
```

Comment:
```suggestion
                    var commandParamsObjWriter = new MonoBinaryWriter();
                    commandParamsObjWriter.Write(0); //param count
                    var retMethod = await context.SdbAgent.InvokeMethod(commandParamsObjWriter.GetParameterBuffer(), methodId, "methodRet", token);
```

Looks like a small error rebasing.

Reply:
Thank you!